File: src/main/java/app/askresume/domain/member/constant/MemberType.java
package app.askresume.domain.member.constant;

import java.util.Arrays;
import java.util.List;

public enum MemberType {

    LOCAL, GOOGLE;

    public static MemberType from(String type) {
        return MemberType.valueOf(type.toUpperCase());
    }

    public static boolean isMemberType(String type) {
        List<MemberType> memberTypes = Arrays.stream(MemberType.values())
                .filter(memberType -> memberType.name().equals(type)).toList();
        return memberTypes.size() != 0;
    }

}

File: src/main/kotlin/app/askresume/domain/member/constant/MemberType.kt
package app.askresume.domain.member.constant

enum class MemberType {

    GOOGLE,
    LINKED_IN,
    ;

    companion object {
        fun from(type: String): MemberType {
            return MemberType.valueOf(type.uppercase())
        }

        fun isMemberType(type: String): Boolean {
            return MemberType.values().any { it.name == type.uppercase() }
        }
    }

}